shpool/tests/attach.rs

```diff
@@ -1287,6 +1287,9 @@ fn dynamic_config_change() -> anyhow::Result<()> {
         let config_contents = config_tmpl.replace("REPLACE_ME", "NEW_VALUE");
         fs::write(&config_file, config_contents)?;
 
+        // Wait for longer than debounce time
+        thread::sleep(time::Duration::from_secs(2));
```

Comment:
If we can, it would be nice to avoid this sleep. To help with this, I wrote the test_hooks subsystem that allows you to emit events from within the daemon code via a macro that gets compiled out when not compiled in test mode, then subscribe to those events within tests. https://github.com/shell-pool/shpool/blob/fef785abbf17dc4e4507dea7273cf52f95d92563/shpool/tests/attach.rs#L121 is an example of how to make the waiter for events in the test.

That might be able to help here.

Reply:
Cool! Thanks for the pointer. Let me take a look and see how the hooks can fit in.